debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresTemporalPrecisionHandlingIT.java

```diff
@@ -0,0 +1,411 @@
+        stopConnector();
+    }
+
+    @Test
+    public void shouldConvertTemporalsMicroseconds() throws Exception {
+        Testing.Print.disable();
+        final PostgresConnectorConfig config = new PostgresConnectorConfig(TestHelper.defaultConfig()
+                .with(PostgresConnectorConfig.INCLUDE_UNKNOWN_DATATYPES, true)
+                .with(PostgresConnectorConfig.SCHEMA_INCLUDE_LIST, "temporaltype")
+                // .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                .with(PostgresConnectorConfig.TIME_PRECISION_MODE, TemporalPrecisionMode.MICROSECONDS)
+                .build());
+        start(PostgresConnector.class, config.getConfig());
+        assertConnectorIsRunning();
+
+        // wait for snapshot completion
+        TestHelper.execute("""
+                INSERT INTO temporaltype.test_data_types
+                VALUES (1 , NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL );""");
+
+        SourceRecords records = consumeRecordsByTopic(1);
+        SourceRecord insertRecord = records.recordsForTopic(TOPIC_NAME).get(0);
+        assertEquals(TOPIC_NAME, insertRecord.topic());
+        VerifyRecord.isValidInsert(insertRecord, "c_id", 1);
+        Struct after = getAfter(insertRecord);
+        assertEquals(after.get("c_id"), 1);
+        assertEquals(after.get("c_date"), null);
+        assertEquals(after.get("c_timestamp0"), null);
+        assertEquals(after.get("c_timestamp1"), null);
+        assertEquals(after.get("c_timestamp2"), null);
+        assertEquals(after.get("c_timestamp3"), null);
+        assertEquals(after.get("c_timestamp4"), null);
+        assertEquals(after.get("c_timestamp5"), null);
+        assertEquals(after.get("c_timestamp6"), null);
+        assertEquals(after.get("c_timestamptz"), null);
+        assertEquals(after.get("c_time"), null);
+        assertEquals(after.get("c_time_whtz"), null);
+        assertEquals(after.get("c_interval"), null);
+
+        TestHelper.execute(
+                """
+                        INSERT INTO temporaltype.test_data_types
+                        VALUES (2 , '{"jfield": 111}'::json , '{"jfield": 211}'::jsonb , '2017-09-15'::DATE , '2019-07-09 02:28:57+01' , '2019-07-09 02:28:57.1+01' , '2019-07-09 02:28:57.12+01' , '2019-07-09 02:28:57.123+01' , '2019-07-09 02:28:57.1234+01' , '2019-07-09 02:28:57.12345+01' , '2019-07-09 02:28:57.123456+01', '2019-07-09 02:28:10.123456+01', '04:05:11 PST', '04:05:11.789', INTERVAL '1' YEAR )
+                        ;""");
+
+        records = consumeRecordsByTopic(1);
+        insertRecord = records.recordsForTopic(TOPIC_NAME).get(0);
+        assertEquals(TOPIC_NAME, insertRecord.topic());
+        VerifyRecord.isValidInsert(insertRecord, "c_id", 2);
+        after = getAfter(insertRecord);
+
+        assertEquals(after.get("c_id"), 2);
+        // '2017-09-15'::DATE
+        // assertEquals(after.get("c_date"), "2017-09-15Z");
```

Comment:
Makes sure no asserts are comented out.

Reply:
Add FIx for date conversion and activated this line.